docs/source/upcoming_changes/README.rst

```diff
@@ -0,0 +1,35 @@
+
+Changelog
+=========
+
+This directory contains "news fragments" which are short files that contain a
+small **ReST**-formatted text that will be added to the next what's new page.
+
+Make sure to use full sentences with correct case and punctuation, and please
+try to use Sphinx intersphinx using backticks. The fragment should have a
+header line and an underline using ``========`` followed by description of
+your user-facing changes as they should appear in the relase notes.
+
+Each file should be named like ``<PULL REQUEST>.<TYPE>.rst``, where
+``<PULL REQUEST>`` is a pull request number, and ``<TYPE>`` is one of:
+
+* ``feature``: Signifying a new feature.
+
+* ``bugfix``: Signifying a bug fix.
+
+* ``doc``: Signifying a documentation improvement.
+
+* ``removal``: Signifying a deprecation or removal of public API.
+
+* ``misc``: A ticket has been closed, but it is not of interest to users.
```

Comment:
These are the default categories of changes provided by `towncrier`. 
(https://towncrier.readthedocs.io/en/stable/tutorial.html#creating-news-fragments)

cc @numba/fpoc; We should be putting together a custom category list of changes that are to be reflected in the auto-generated release notes.

Reply:
Made the following list of categories adapted from NumPy:

```
* ``highlight``: Adds a highlight bullet point to use as a possibly highlight
  of the release.
* ``np_support``: Addition of new NumPy functionality.
* ``deprecation``: Changes existing code to emit a DeprecationWarning.
* ``future``: Changes existing code to emit a FutureWarning.
* ``expired``: Removal of a deprecated part of the API.
* ``compatibility``: A change which requires users to change code and is not
  backwards compatible. (Not to be used for removal of deprecated features.)
* ``cuda``: Changes in the CUDA API.
* ``new_feature``: New user facing features like ``kwargs``.
* ``improvement``: General improvements and edge-case changes which are
  not new features or compatibility related.
* ``performance``: Performance changes that should not affect other behaviour.
* ``change``: Other changes
```

Reply:
@kc611 Suggest raising this at the next maintainer/triage meeting to make sure that this list acceptable to all maintainers, also think that the general form of the release notes needs a general discussion/acceptance from the same group.